models/ed/R/model2netcdf.ED2.R

```diff
@@ -643,11 +647,25 @@ put_T_values <- function(yr, nc_var, out, lat, lon, begins, ends, ...){
   
   
   # ----- define ncdf dimensions
-  
+  # setup output time and time bounds
+  sub_date_range <- seq(begins, by = "day", length.out = length(lubridate::yday(begins):ends))
```

Comment:
Couldn't you simplify `length(yday(begins):ends)` with just `ends - begins`?

Reply:
```
  sub_date_range <- seq(begins, by = "day", length.out = length(lubridate::yday(begins):ends))

> head(sub_date_range)
[1] "2001-07-15" "2001-07-16" "2001-07-17" "2001-07-18" "2001-07-19"
[6] "2001-07-20"

> tail(sub_date_range)
[1] "2001-12-26" "2001-12-27" "2001-12-28" "2001-12-29" "2001-12-30"
[6] "2001-12-31"
```

```
> sub_date_range <- seq(begins, by = "day", length.out = ends-lubridate::yday(begins))
> head(sub_date_range)
[1] "2001-07-15" "2001-07-16" "2001-07-17" "2001-07-18" "2001-07-19"
[6] "2001-07-20"

> tail(sub_date_range)
[1] "2001-12-25" "2001-12-26" "2001-12-27" "2001-12-28" "2001-12-29"
[6] "2001-12-30"
```

So it misses a day

Reply:
One of the crazy issues with "time" in R and python, it seems, is that if you parse a date, say Jan 1 into DOY it will give you 1 but if you provide DOY 1 and parse to date it will give you Jan 2.  In other words, 0 index going from DOY to date but indexed from 1 going back!?  So it makes creating correct date vectors challenging, and as such, I create a date vector and convert to DOY then subtract 1 to be sure...more or less.

Reply:
Re: missing a day, you could probably just do `1 + ends - yday(begins)`, but point taken -- `length(...)` may be clearer.

Re: DOY issues -- Sounds like a pain, but I'm struggling to reproduce. I can go from date to DOY with `lubridate::yday("2018-01-01")`, which correctly returns `1`. How are you going from DOY to date?

Reply:
FYI, `seq` has an `along.with` argument:

```
along.with: take the length from the length of this argument.
```

In other words, I think `seq(x, length.out = length(y:z))` is the same as `seq(x, along.with = y:z)`.

Reply:
@ashiklom now go from DOY 1 and see what it gives you as a date using as.Date(1, origin="2018-01-01").  My guess is day 2

Reply:
For example

```
> lubridate::yday("2018-01-01")
[1] 1
> as.Date(1, origin="2018-01-01")
[1] "2018-01-02"
```

Reply:
```
> as.Date(0, origin="2018-01-01")
[1] "2018-01-01"
```
We often grab the "y" or year variable and use that as origin so as such I like to use the time functions and then finally convert to 0 index once I have the correct date vectors, etc rather than messing with origin

Reply:
Ah, OK. Strictly speaking, that makes sense since dates are defined as `origin + X days` (and POSIXct datetimes are defines as `origin + X seconds`), but I see how introduces confusion.